### PR TITLE
Update deps: k8s, ginkgo, gomega, logr

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Update main dependencies, in preparation for 0.2 release.

Remove custom gomega matcher in favor of `gomega.BeComparableTo` onsi/gomega#216

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,21 +3,21 @@ module sigs.k8s.io/kueue
...
+	go.uber.org/zap v1.22.0
+	k8s.io/api v0.24.3
+	k8s.io/apimachinery v0.24.3
+	k8s.io/client-go v0.24.3
+	k8s.io/component-base v0.24.3
+	k8s.io/component-helpers v0.24.3
+	k8s.io/klog/v2 v2.60.1
 	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9
-	sigs.k8s.io/controller-runtime v0.11.1
+	sigs.k8s.io/controller-runtime v0.12.3
```
Comment (ahg-g) [inline go.mod:20]: was this the only block to be updated manually for dependency update?

Comment (alculquicondor) [reply go.mod:20]: `require`? yes. The rest was done by `go mod tidy`

**Pull request merged**